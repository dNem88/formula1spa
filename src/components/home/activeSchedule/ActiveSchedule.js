import React from 'react'
import styles from './activeSchedule.module.css'
import Flag from '../flag/Flag'

function ActiveSchedule(props) {
    let raceTitle = `formula 1 ${props.raceName} ${props.season}`.toUpperCase();

    return (
        <div className={styles['wrapper']}>
            <div className={styles['content-container']}>
                <Flag classname={'left-flag'} imageBase={props.imageBase}/>
                <h5 className={styles.country}>{props.country}</h5>
                <h2 className={styles["gp-name"]}>{raceTitle}</h2>
                <h6 className={styles['weekend-duration']}>{props.weekend}</h6>
            </div>
            <div className={styles['clock-banner-container']}>

            </div>
            <div className={styles['table-container']}>

            </div>
        </div>
    )
}

export default ActiveSchedule