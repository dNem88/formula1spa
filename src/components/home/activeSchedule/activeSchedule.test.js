import React from 'react'
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'
import ActiveSchedule from './ActiveSchedule'

let mockProps = {
    raceName: 'test-race',
    season: '2022',
    classname: 'test-class',
    imageBase: 'test-base',
    country: 'test-country',
    raceTitle: 'test-title',
    weekend: 'test-weekend',

}
describe('Test ActiveSchedule' , () => {
    it('Should render properly', () => {
        let activeSchedule = renderer.create(
            <MemoryRouter>
                <ActiveSchedule {...mockProps}/>
            </MemoryRouter>
        ).toJSON()
        expect(activeSchedule.type).toBe('div')
        expect(activeSchedule.props.className).toBe('wrapper')
        expect(activeSchedule.children.length).toBe(3)
        expect(activeSchedule.children[0].type).toBe('div')
        expect(activeSchedule.children[0].props.className).toBe('content-container')
        expect(activeSchedule.children[1].type).toBe('div')
        expect(activeSchedule.children[1].props.className).toBe('clock-banner-container')
        expect(activeSchedule.children[2].type).toBe('table')
        expect(activeSchedule.children[2].props.className).toBe('race-time-container')
        let contentContainer = activeSchedule.children[0]
        expect(contentContainer.children.length).toBe(4)
        expect(contentContainer.children[0].type).toBe('div')
        expect(contentContainer.children[0].props.className).toBe('test-class')
        expect(contentContainer.children[1].type).toBe('h5')
        expect(contentContainer.children[1].props.className).toBe('country')
        expect(contentContainer.children[1].children[0]).toBe('test-country')
        expect(contentContainer.children[2].type).toBe('h2')
        expect(contentContainer.children[2].props.className).toBe('gp-name')
        expect(contentContainer.children[2].children[0]).toBe(`formula 1 ${mockProps.raceName} ${mockProps.season}`.toUpperCase())
        expect(contentContainer.children[3].type).toBe('h6')
        expect(contentContainer.children[3].props.className).toBe('weekend-duration')
        expect(contentContainer.children[3].children[0]).toBe('test-weekend')
        let clockContainer = activeSchedule.children[1]
        let tableContainer = activeSchedule.children[2]
    })
})