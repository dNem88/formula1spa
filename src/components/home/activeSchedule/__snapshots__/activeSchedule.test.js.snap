// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ActiveSchedule Should render properly 1`] = `
<div
  className="wrapper"
>
  <div
    className="content-container"
  >
    <div
      className="test-class"
    >
      <img
        alt="flag"
      />
    </div>
    <h5
      className="country"
    >
      test-country
    </h5>
    <h2
      className="gp-name"
    >
      FORMULA 1 TEST-RACE 2022
    </h2>
    <h6
      className="weekend-duration"
    >
      test-weekend
    </h6>
  </div>
  <div
    className="clock-banner-container"
  />
  <div
    className="table-container"
  />
</div>
`;
